jenkins/parser/jenkins-parser-job.py

```diff
@@ -0,0 +1,623 @@
+    with open(rotation_file_path, "r") as rotation_file:
+        old_running_builds_object = json.load(rotation_file)
+        old_running_builds_dict = old_running_builds_object["runningBuilds"]
+
+    if job_to_retry not in old_running_builds_dict:
+        old_running_builds_dict[job_to_retry] = dict()
+        old_running_builds = dict()
+    else:
+        old_running_builds = old_running_builds_dict[job_to_retry]
+
+    # Update value of old running builds in the original object to store it again
+    for build_number in current_running_builds:
+        if (
+            build_number
+            not in old_running_builds_object["runningBuilds"][job_to_retry].keys()
+        ):
+            old_running_builds_object["runningBuilds"][job_to_retry][build_number] = ""
+
+    with open(rotation_file_path, "w") as rotation_file:
+        json.dump(old_running_builds_object, rotation_file)
+
+    return current_running_builds, old_running_builds
+
+
+def get_new_running_builds(job_dir, last_processed_log):
+    """Get list of new running builds that have been started after the last processed log."""
+    return [
+        dir.name
+        for dir in os.scandir(job_dir)
+        if dir.name.isdigit()
+        and int(dir.name) > int(last_processed_log)
+        and os.path.isfile(
+            functools.reduce(os.path.join, [job_dir, dir.name, "build.xml"])
+        )
+        and not grep(
+            functools.reduce(os.path.join, [job_dir, dir.name, "build.xml"]), "<result>"
+        )
+    ]
+
+
+def check_and_trigger_action(build_to_retry, job_dir, job_to_retry, error_list_action):
+    """Check failed build logs and trigger the appropiate action if a known error is found."""
+    build_dir_path = os.path.join(job_dir, build_to_retry)
+    log_file_path = os.path.join(build_dir_path, "log")
+    envvars_file_path = os.path.join(build_dir_path, "injectedEnvVars.txt")
+
+    # TODO: Try not to load everything on memory
+    text_log = open(log_file_path, errors="ignore")
+    lines = text_log.readlines()
+    text_log.close()
+
+    job_url = (
+        "https://cmssdt.cern.ch/jenkins/view/All/job/"
+        + job_to_retry
+        + "/"
+        + build_to_retry
+    )
+
+    print("Parsing build #" + build_to_retry + " (" + job_url + ") ...")
+
+    regex_flag = 0
+    for error_and_action in error_list:
+        regex_and_action = error_and_action.split(" - ")
+        regex = regex_and_action[0]
+        action = regex_and_action[1]
+        for line in reversed(lines):
+            if re.search(regex, line):
+                print(
+                    "... Found message "
+                    + regex
+                    + " in "
+                    + log_file_path
+                    + ". Taking action ..."
+                )
+                if action == "retryBuild":
+                    trigger_retry = (
+                        "ssh -o IdentitiesOnly=yes -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no "
```

Comment:
@aandvalenzuela , please use `JENKINS_CLI_CMD` env variable which should be set to something like `ssh -o <ssh-options> -i <path to ssh key> -l <local user> -p <port> <host>`

Reply:
this is also used in few place below, so update all to those to use `JENKINS_CLI_CMD` env